python/doc/examples/meta_modeling/kriging_metamodel/plot_kriging_beam_arbitrary_trend.py

```diff
@@ -92,10 +92,18 @@
 # In order to create the kriging metamodel, we first select a constant trend with the `ConstantBasisFactory` class. Then we use a squared exponential covariance model. Finally, we use the `KrigingAlgorithm` class to create the kriging metamodel, taking the training sample, the covariance model and the trend basis as input arguments. 
 
 # %%
-covarianceModel = ot.SquaredExponential([1.]*dimension, [1.0])
+covarianceModel = ot.SquaredExponential(X_train.getMax(), [1.0])
+
+# %%
+# We need to manually define sensible optimization bounds.
+# Note that since the amplitude parameter is computed analytically (this is possible when the output dimension is 1), we only need to set bounds on the scale parameter.
+
+# %%
+scaleOptimizationBounds = ot.Interval([1.0, 1.0, 1.0, 1.0e-10], [1.0e11, 1.0e3, 1.0e1, 1.0e-5])
```

Comment:
could you explain why we need to set bounds now ?
I dont recall Regis proposal (I dont have his code anymore), but I think it was trying to set up bounds

Reply:
Actually, we always should have set bounds in [cantilever beam](https://openturns.github.io/openturns/1.16/usecases/use_case_cantilever_beam.html) examples because the moment of intertia (the last component) follows a Beta distribution with upper bound 1.7e-7.

The GLM optimization lower bound is by default 0.01, whereas the upper bound is computed from the input sample.
In this case, the default optimization upper bound is actually *lower* than the default optimization lower bound, so the optimization bounds define an empty set.

In the current master, when the starting point(s) are outside the optimization bounds (which is automatically the case when they define an empty set), the optimization silently fails. With the checks I have implemented, GeneralLinearModelAlgorithm throws if the optimization bounds define an empty Interval. That is why I need to define sensible optimization bounds to make `plot_kriging_beam_arbitrary_trend.py` run.